src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,959 @@
+
+            // prevent infinite recursion:
+            Debug.Assert(!ReferenceEquals(copyFrom, this));
+
+            bool bagCreatedOnThisThread;
+            if ((object)copyFrom != null)
+            {
+                var attributesBag = forReturnType ? copyFrom.GetReturnTypeAttributesBag() : copyFrom.GetAttributesBag();
+                bagCreatedOnThisThread = Interlocked.CompareExchange(ref lazyCustomAttributesBag, attributesBag, null) == null;
+            }
+            else if (forReturnType)
+            {
+                bagCreatedOnThisThread = LoadAndValidateAttributes(
+                    this.GetReturnTypeAttributeDeclarations(),
+                    ref lazyCustomAttributesBag,
+                    symbolPart: AttributeLocation.Return);
+            }
+            else
+            {
+                bagCreatedOnThisThread = LoadAndValidateAttributes(this.GetAttributeDeclarations(), ref lazyCustomAttributesBag);
+            }
+
+            if (bagCreatedOnThisThread)
+            {
+                NoteAttributesComplete(forReturnType);
+
+                if (!forReturnType
+                    && IsExtern
+                    && !IsAbstract
+                    && !this.IsPartialMethod()
+                    && GetBinderContextNode() is null
+                    && lazyCustomAttributesBag.Attributes.IsEmpty
+                    && !this.ContainingType.IsComImport)
+                {
+                    var diagnostics = DiagnosticBag.GetInstance();
+                    // external method with no attributes
+                    var errorCode = (this.MethodKind == MethodKind.Constructor || this.MethodKind == MethodKind.StaticConstructor) ?
+                        ErrorCode.WRN_ExternCtorNoImplementation :
+                        ErrorCode.WRN_ExternMethodNoImplementation;
+                    diagnostics.Add(errorCode, this.Locations[0], this);
+                    AddDeclarationDiagnostics(diagnostics);
+                    diagnostics.Free();
+                }
+            }
+
+            return lazyCustomAttributesBag;
+        }
+
+        /// <summary>
+        /// Called when this thread loaded the method's attributes. For method symbols with completion state.
+        /// </summary>
+        protected virtual void NoteAttributesComplete(bool forReturnType) { }
+
+        /// <summary>
+        /// Gets the attributes applied on this symbol.
+        /// Returns an empty array if there are no attributes.
+        /// </summary>
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            return this.GetAttributesBag().Attributes;
+        }
+
+        /// <summary>
+        /// Gets the attributes applied on the return value of this method symbol.
+        /// Returns an empty array if there are no attributes.
+        /// </summary>
+        public override ImmutableArray<CSharpAttributeData> GetReturnTypeAttributes()
```

Comment:
>override [](start = 15, length = 8)

`sealed`?

Reply:
Many of these members are overridden by `SynthesizedMethodBaseSymbol`, but I will make a pass to make override members sealed where possible.

---
In reply to: [353404454](https://github.com/dotnet/roslyn/pull/40011#discussion_r353404454) [](ancestors = 353404454)